components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/IdentityManagementEndpointUtil.java

```diff
@@ -513,12 +522,34 @@ public static String getURLEncodedCallback(String callbackUrl) throws URISyntaxE
         return encodedQueryMap;
     }
 
+    /**
+     * Construct the server URL depending on the path and the resource name.
+     *
+     * @param tenantDomain tenant Domain.
+     * @param context      context of the url.
+     * @return endpoint url.
+     */
+    public static String buildServerUrl(String tenantDomain, String context) throws URLBuilderException {
+
+        String serverUrl = IdentityManagementServiceUtil.getInstance().getContextURL();
+        if (StringUtils.isBlank(serverUrl)) {
+            return ServiceURLBuilder.create().addPath(context).build().getAbsoluteInternalURL();
+        } else if (StringUtils.isBlank(tenantDomain)) {
+            return serverUrl + context;
+        } else {
+            return serverUrl + "/t/" + tenantDomain + context;
+        }
```

Comment:
Rather than depending on the tenant domain passed and addressing the legacy mode, we should rely only on the configuration.
Because, later some might use the same method in an unintended way

Reply:
May be you case pass some fields to decide whether to append tenant in legacy mode or not

Reply:
From the way this API is being utilized below, seems it is used to build the url at non tenanted mode.
I don't think we should make this public.
Unless we can handle this from one public API let's rename this method to denote the case that this is to be invoked at non tenanted mode.

Reply:
Changed the method as discussed.